### DISCUSSION TITLE
EMEI Or Other Device Unique ID

### DISCUSSION THREAD

**Discussion by adrianb9** [owntracks/android#1377] [Ideas & New Features]
Hi,

is it possible to have the EMEI or another ID that is linked to the phone. I want to be able to recognise the device that is sending via a HTTP post and do not want the user to be able to change it.

Thanks for a great App.

Adrian

**Comment by jpmens** [comment]
I think we decided quite a while back to not do that for privacy reasons, though it might actually be that we have no access to the IMEI.

Would setting a topic (or TID) and then locking the preferences be a solution?

**Reply by adrianb9** [reply]
Hi Jan-Piet,
Yes that could work. Can this be done already or is it a feature request ?

Thanks
Adrian

**Comment by jpmens** [comment]
Sorry, Adrian; that might be an iOS only feature.

@growse can you chip in, please?

**Reply by growse** [reply]
You can set the deviceId and trackerId in Android.

**Reply by jpmens** [reply]
@growse my question was about locking; can prefs currently be locked in Android?

**Reply by growse** [reply]
No. Not even sure how that could be done - any setting is just stored in a preference, and these are just XML files on the device.

If you want to hard bake something in, I think the right way would be to fork and make your own build.
